build.gradle

```diff
@@ -1521,6 +1522,91 @@ project(':group-coordinator') {
   srcJar.dependsOn 'processMessages'
 }
 
+project(':test-common') {
+  base {
+    archivesName = "kafka-test-common"
+  }
+
+  dependencies {
+    implementation libs.scalaJava8Compat
+    implementation project(':core')
+    implementation project(':metadata')
+    implementation project(':server')
+    implementation project(':raft')
+    implementation project(':storage')
+    implementation project(':server-common')
+    implementation libs.slf4jApi
+    testImplementation libs.junitJupiter
+    testImplementation libs.mockitoCore
+    testRuntimeOnly libs.slf4jReload4j
+    testRuntimeOnly libs.junitPlatformLanucher
```

Comment:
Will this configuration be inherited by modules now depending on `test-common`? Or do they still need to explicitly depend on these runtime deps?

If these are not brought over transitively, then we may consider a follow-up of adding a "test-common-runtime" module that includes these things so we can enforce uniform dependencies across the modules (and reduce copy/pasting of deps).

Reply:
Junit 5 recommend to explicitly add the dependency on `junit-platform-launcher`. To quote:
> If using JUnit 5, an explicit runtimeOnly dependency on junit-platform-launcher is required in addition to the existing implementation dependency on the test engine.

https://docs.gradle.org/8.3/userguide/upgrading_version_8.html#manually_declaring_dependencies

Maybe we can add the junit dependencies explicitly on the `allprojects` block for all sub-modules. However, that may complicate the build script.

Reply:
What I was thinking was replacing all of the 
```
dependencies {
    ...
    testRuntimeOnly libs.slf4jReload4j
    testRuntimeOnly libs.junitPlatformLanucher
}
```

with 
```
dependencies {
    ...
    testRuntimeOnly project(":test-common:test-common-runtime")
}
```

where ":test-common-runtime" has

```
dependencies {
    runtimeOnly libs.slf4jReload4j
    runtimeOnly libs.junitPlatformLanucher
}
```

Reply:
I'm not sure if it's worth creating a new module. Perhaps we could just declare a variable instead?
```
ext {
  runtimeLibs = [
    libs.slf4jReload4j,
    libs.junitPlatformLanucher,
  ]

testRuntimeOnly runtimeLibs
```

Reply:
open https://issues.apache.org/jira/browse/KAFKA-17685